common.js

```diff
@@ -69,7 +69,14 @@ function generateFinalBasename (opts) {
 }
 
 function generateFinalPath (opts) {
-  return path.join(opts.out || process.cwd(), generateFinalBasename(opts))
+  if (opts.out) {
+    if (typeof opts.out === 'string') {
+      return path.join(opts.out, generateFinalBasename(opts))
+    } else {
+      console.warn('Output dir must be of type string, fallback to use current working dir instead.')
```

Comment:
I don't think this is the warning we want to send if `opts.out` isn't specified.